### PR TITLE
Close: Add Lead External ID and Lead Custom Fields

### PR DESCRIPTION
**Pull request by froxCZ**
Add Lead External ID and Lead Custom Fields to Close's CreateUpdateContactAndLead action.

## Testing

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by dlasky** [comment]
hey @froxCZ i'm taking a look!

File: packages/destination-actions/src/destinations/close/index.ts [hunk 1]
```diff
@@ -17,6 +17,20 @@ const destination: DestinationDefinition<Settings> = {
         type: 'string',
         required: true
       },
+      lead_custom_field_id_for_company_id: {
```
Comment (dlasky) [inline packages/destination-actions/src/destinations/close/index.ts:20]: is there ever a case where a user might want this to be a different value depending on the segment event type or other filters? (just wondering if this should be in the action fields)

Comment (froxCZ) [reply packages/destination-actions/src/destinations/close/index.ts:20]: Very unlikely, but I will double check with the team. Nevertheless you can go ahead and merge as it is now. If we decide to move it to the Action level we will do so before public release if that's ok.

Comment (dlasky) [reply packages/destination-actions/src/destinations/close/index.ts:20]: no worries, like i said not a blocker. thank you!

**Pull request merged**